+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+classdef Gcs < vdx.problems.Mpcc
+    properties
+        model
+        dcs
+        opts
+
+        populated
+    end
+
+    methods
+        function obj = Gcs(dcs, opts)
+            obj = obj@vdx.problems.Mpcc();
+            obj.model = dcs.model;
+            obj.dcs = dcs;
+            obj.opts = opts;
+        end
+
+        function create_variables(obj)
+            dims = obj.dcs.dims;
+            dcs = obj.dcs;
+            model = obj.model;
+            opts = obj.opts;
+
+            % Parameters
+            obj.p.rho_h_p = {{'rho_h_p',1}, 1};
+            obj.p.rho_terminal_p = {{'rho_terminal_p',1}, 1};
+            obj.p.T = {{'T',1}, opts.T};
+            obj.p.p_global = {model.p_global, model.p_global_val};
+
+
+            % 0d vars: Variables which only exist once globally. 
+            obj.w.v_global = {{'v_global',dims.n_v_global}, model.lbv_global, model.ubv_global, model.v0_global};
+            if opts.use_speed_of_time_variables && ~opts.local_speed_of_time_variable
+                obj.w.sot = {{'sot', 1}, opts.s_sot_min, opts.s_sot_max, opts.s_sot0};
+            end
+            if opts.time_optimal_problem
```

Comment:
please add a time optimal pds example, just to verify the correctness of this (e.g. you can use the standard simulation pds, add some control in 2d, and force that the "particle" ends somewhere on the constraint. for instance the control could change the vector field such that it does not leave the "sliding mode" but stays on it

Reply:
okay, i see that the msv example is a time optimal one.